src/Bicep.Core/Semantics/NameBindingVisitor.cs

```diff
@@ -68,6 +72,50 @@ public override void VisitVariableAccessSyntax(VariableAccessSyntax syntax)
             this.bindings.Add(syntax, symbol);
         }
 
+        public override void VisitResourceAccessSyntax(ResourceAccessSyntax syntax)
+        {
+            base.VisitResourceAccessSyntax(syntax);
+
+            // we need to resolve which resource delaration the LHS is pointing to - and then 
+            // validate that we can resolve the name.
+            if (!this.bindings.TryGetValue(syntax.BaseExpression, out var symbol))
+            {
+                // This can occur when parsing errors happen like `a:`

```

Comment:
>This [](start = 19, length = 4)

The body of this branch isn't doing anything. Is that intentional?

Reply:
Good catch, it was intentional to do nothing, but the it's better just not have the block this in case. Cleaned this up.